bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java

```diff
@@ -34,7 +39,41 @@
 public abstract class AbstractCloudBigtableTableDoFn<In, Out> extends DoFn<In, Out> {
   private static final long serialVersionUID = 1L;
 
-  private static final CloudBigtableConnectionPool pool = new CloudBigtableConnectionPool();
+  static final CloudBigtableConnectionPool pool = new CloudBigtableConnectionPool();
+
+  public static void logException(
+      Logger log, String context, RetriesExhaustedWithDetailsException exception) {
+    log.warn("For context {}: exception occured during bulk writing: {}", context,
```

Comment:
RetriesExhaustedWithDetailsException could be for bulk reads as well.  I'll just say RetriesExhaustedWithDetailsException occurred...